torch/csrc/distributed/c10d/exception.h

```diff
@@ -9,37 +9,22 @@
 #include <stdexcept>
 
 #include <c10/macros/Macros.h>
+#include <c10/util/Exception.h>
 
-namespace c10d {
-
-class TORCH_API C10dError : public std::runtime_error {
- public:
-  using std::runtime_error::runtime_error;
-
-  C10dError(const C10dError&) = default;
+#define C10D_THROW_ERROR(err_type, msg) \
```

Comment:
whats the guidance on when to use `C10_THROW_ERROR` vs `C10D_THROW_ERROR` as in some parts I see it is intermingled. Could this be written as a comment?

Reply:
Sure, will add a comment. I guess `C10D_THROW_ERROR` is for c10d:: error types and `C10_THROW_ERROR` is for c10:: error types.